homeassistant/components/media_source/local_source.py

```diff
@@ -54,13 +35,38 @@ def __init__(self, hass: HomeAssistant):
         self.hass = hass
 
     @callback
-    def async_full_path(self, source_dir_id, location) -> Path:
+    def async_full_path(self, source_dir_id, location) -> str:
         """Return full path."""
-        return self.hass.config.path("media", location)
+        return str(Path(self.hass.config.media_dirs[source_dir_id], location))
+
+    @callback
+    def async_parse_identifier(self, item: MediaSourceItem) -> Tuple[str, str]:
+        """Parse identifier."""
+        if not item.identifier:
+            source_dir_id = ""
+            location = ""
+
+        else:
+            source_dir_id, location = (
+                item.identifier.replace("/local_source/", "", 1)
```

Comment:
The web url should not be part of the passed in identifier. Why do we have to replace it here?

The code that passes it _with_ url should strip it out.

Reply:
You are right, this is actually a left-over from an earlier attempt to get the additional path in the URLs working from the frontend. Later replaced by the handling now already here.

Removed it.